File: Core/actor/actorbase.cpp
#include "actorbase.h"

#include <QDebug>

QLatin1String ActorBase::PROPERTY_IS_ASYNC = QLatin1String("is_async");
QLatin1String ActorBase::PROPERTY_PRIORITY = QLatin1String("priority");

ActorBase::ActorBase() : ValueBase() {
}

ActorBase::ActorBase(ValueGroup *valueGroup, QString id, VALUE_TYPE valueType, QVariant::Type typeHint, QObject *parent) : ValueBase(valueGroup, id, valueType, true, typeHint, parent)
{

}

void ActorBase::triggerCmd(actor::ACTOR_CMDS cmd, QString reason) {
    iDebug() << Q_FUNC_INFO << cmd << reason;

    if (cmdSupported(cmd)) {
        _triggerCmd(cmd);
        Q_EMIT(cmdTriggered(cmd));
    } else {
        iWarning() << "Cmd not supported" << cmd;
    }
}

/*
void ActorBase::triggerCmd(int cmd) {
    iDebug() << Q_FUNC_INFO << cmd;
    triggerCmd(static_cast<ACTOR_CMDS>(cmd));
}
*/

void ActorBase::_triggerCmd(actor::ACTOR_CMDS cmd) {

}

ActorBase* ActorBase::withPriority(int priority) {
    this->m_priority = priority;
    return this;
}

int ActorBase::priority() {
    return m_priority;
}

File: AndroidApp/OshCore/src/main/java/com/osh/actor/ActorBase.java
package com.osh.actor;

import com.osh.value.ValueBase;
import com.osh.value.ValueGroup;
import com.osh.value.ValueType;

import java.util.Map;

public abstract class ActorBase<VALUE_TYPE extends ActorBase, NATIVE_TYPE> extends ValueBase<VALUE_TYPE, NATIVE_TYPE> {

	protected Map<String, Object> config;
	public abstract boolean cmdSupported(ActorCmds cmd);
    protected abstract void _triggerCmd(ActorCmds cmd);

	public ActorBase(ValueGroup valueGroup, String id, ValueType valueType) {
		super(valueGroup, id, valueType);
	}

	public ActorBase() {
	}
	
	public void triggerCmd(ActorCmds cmd, String reason) {
	    if (cmdSupported(cmd)) {
	        _triggerCmd(cmd);
	    }
	}


}
